src/sdk/refundPegin.ts

```diff
@@ -0,0 +1,48 @@
+import type { Quote } from '../api'
+import type { FlyoverSDKContext } from '../utils/interfaces'
+import { isPeginRefundable } from './isPeginRefundable'
+import { getRawTxWithoutWitnesses } from '../bitcoin/utils'
+import pmtBuilder from '@rsksmart/pmt-builder'
+
+export interface RefundPeginParams {
+  quote: Quote
+  providerSignature: string
+  userBtcTransactionHash: string
+  flyoverContext: FlyoverSDKContext
+}
+
+export async function refundPegin (
+  params: RefundPeginParams
+): Promise<string> {
+  const { quote, providerSignature, userBtcTransactionHash, flyoverContext } = params
+
+  const isRefundable = await isPeginRefundable({
+    quote,
+    providerSignature,
+    btcTransactionHash: userBtcTransactionHash,
+    flyoverContext
+  })
+
+  if (!isRefundable.isRefundable) {
+    throw isRefundable.error
```

Comment:
if I recall correctly, the FlyoverErrors objects are not errors right? maybe it would be better to throw an actual error here